awx/ui/client/src/instance-groups/container-groups/add-container-group.controller.js

```diff
@@ -0,0 +1,68 @@
+function AddContainerGroupController(ToJSON, $scope, $state, models, strings, i18n, DataSet) {
+  const vm = this || {};
+  const {
+    instanceGroup,
+    credential
+  } = models;
+
+  vm.mode = 'add';
+  vm.strings = strings;
+  vm.panelTitle = strings.get('state.ADD_CONTAINER_GROUP_BREADCRUMB_LABEL');
+  vm.lookUpTitle = strings.get('container.LOOK_UP_TITLE');
+
+  vm.form = instanceGroup.createFormSchema('post');
+  vm.form.name.required = true;
+
+  vm.form.credential = {
+    type: 'field',
+    label: i18n._('Credential'),
+    id: 'credential'
+  };
+  vm.form.credential._resource = 'credential';
+  vm.form.credential._route = "instanceGroups.addContainerGroup.credentials";
+  vm.form.credential._model = credential;
+  vm.form.credential._placeholder = strings.get('container.CREDENTIAL_PLACEHOLDER');
+  vm.form.credential.required = true;
+
+  vm.form.extraVars = extraVarsDetails();
```

Comment:
see other comment about  removing the `extraVarsDetails` function and changing these to
```js
vm.form.extraVars = {
    label: strings.get('container.POD_SPEC_LABEL'),
    value: EditContainerGroupDataset.data.pod_spec_override,
    name: 'extraVars'
};
```